### DISCUSSION TITLE
how not to use a root owned mount as data volume?

### DISCUSSION THREAD

**Discussion by DuelCoreF18** [abraunegg/onedrive#1911] [Q&A] [answered]
Can you provide some best practice on how to use a share from my TrueNas VM as my data volume? I am really confused on how to avoid the "without having the target mounted by 'root'" issue when every folder in my mounted cifs share on my Docker VM shows as owned by root.

**Comment by abraunegg** [comment, marked as answer]
@DuelCoreF18 
Without knowing your entire setup - look at the following:
* You need to create users on your TrueNAS VM - so that your user has read|write permissions to that shared folder
* When mounting that folder on Linux you need to specify those credentials - and those credentials map back to a user who is mapped to your local Linux users & groups
  * Example: `mount.cifs //192.168.1.9/storage-photo /mnt/windows-storage -o rw,username=my_user,password=my_password`

An alternative would be to use NFS rather than CIFS .. but I am unsure if TrueNAS can share a data volume as CIFS and NFS at the same time ..
